### PR TITLE
[To Feature] DESENG-501 Added new content tabs to engagement form

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-501

*Description of changes:*
- Front end changes
- Added new content tabs (summary and custom) to engagement form.
- Moved engagement page content and survey block to summary tab.
- Made summary tab default on engagement creation and added logic to add a new custom tab with a list of provided icons.
- Custom tab to have a default template with a text editor.
- Added logic to edit or delete a tab.
- Modifications were made to maintain the existing functionality for content-related tasks.

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/services/engagementSummaryService/index.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { EngagementSummaryContent } from 'models/engagementSummaryContent';
+import Endpoints from 'apiManager/endpoints';
+import { replaceAllInURL, replaceUrl } from 'helper';
+
+export const getSummaryContent = async (contentId: number): Promise<EngagementSummaryContent[]> => {
+    const url = replaceUrl(Endpoints.EngagementSummaryContent.GET, 'content_id', String(contentId));
+    if (!contentId || isNaN(Number(contentId))) {
+        return Promise.reject('Invalid content Id ' + contentId);
+    }
+    const response = await http.GetRequest<EngagementSummaryContent[]>(url); // Notice the change here
```
Comment (Baelx) [inline met-web/src/services/engagementSummaryService/index.ts:11]:
```suggestion
    const response = await http.GetRequest<EngagementSummaryContent[]>(url);
```

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/SummaryTabContent.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+                <MetDescription>This is the main content of the engagement page.</MetDescription>
+
+                <div style={{ position: 'relative' }}>
+                    <RichTextEditor
+                        setRawText={handleContentChange}
+                        handleEditorStateChange={handleRichContentChange}
+                        initialRawEditorState={initialRichContent || ''}
+                    />
+                    {/* Overlay to prevent user interaction */}
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/SummaryTabContent.tsx:55]: Appreciate this comment to explain what's happening in the code!

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/ContentTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    const handleDeleteTab = async (index: number) => {
+        const tab_id = contentTabs[index].id;
+        if (!tab_id) {
+            return;
+        }
+        await deleteEngagementContent(savedEngagement.id, tab_id);
+        await fetchEngagementContents();
+        setTabIndex(0);
+
+        // Check if the deleted tab was a custom tab and reset customTabAdded state
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/ContentTabs.tsx:47]: Useful comment!

**Review by Baelx — CHANGES_REQUESTED** [review]
Great work! Could I ask that you:
- Please review the comments in the code and remove those that don't add any new information or that don't describe what's happening in the code.
- We should try to keep code as DRY as possible. Is there any way to merge `AddContentTabModal` and `EditContentTabModal`. We can discuss further if you'd like!

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/CustomTabContent.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            direction="row"
+            justifyContent="flex-start"
+            alignItems="baseline"
+            rowSpacing={{ xs: 1, sm: 0 }}
+            columnSpacing={2}
+        >
+            <Grid item xs={12}>
+                <MetLabel>Engagement - Page Custom Content</MetLabel>
+
+                <MetDescription>This is the additional content of the engagement page.</MetDescription>
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/CustomTabContent.tsx:38]: This is just a description that appears to the engagement creator. We had a similar one on engagement content so I kept it for this one as well. Please let me know if we can remove this.
image

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/ContentTabs.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+export const ContentTabs: React.FC = () => {
+    const { fetchEngagementContents, contentTabs, setContentTabs, savedEngagement } = useContext(ActionContext);
+    const { isSummaryContentsLoading } = useContext(EngagementContentContext);
+    const [tabIndex, setTabIndex] = useState(0);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+    const [selectedTabType, setSelectedTabType] = useState('');
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const [customTabAdded, setCustomTabAdded] = useState(false); // Define state to track whether a custom tab has been added
+    const handleChange = (_event: React.SyntheticEvent, newValue: number) => {
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/ContentTabs.tsx]: Updated, thanks.

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/ContentTabs.tsx [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+                        <EditContentTabModal
+                            open={isEditModalOpen}
+                            updateModal={handleEditModalClose}
+                            tabs={contentTabs}
+                            setTabs={setContentTabs}
+                            selectedTabIndex={tabIndex}
+                        />
+                        <Box mt={3}>
+                            <MetTabPanel value={tabIndex.toString()}>
+                                {/* Conditional rendering */}
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/form/EngagementFormTabs/EngagementContent/ContentTabs.tsx]: Thanks for noticing, I have removed this.

File: met-web/src/services/engagementContentService/index.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+import { EngagementContent } from 'models/engagementContent';
+import Endpoints from 'apiManager/endpoints';
+import { replaceAllInURL, replaceUrl } from 'helper';
+
+export const getEngagementContent = async (engagementId: number): Promise<EngagementContent[]> => {
+    const url = replaceUrl(Endpoints.EngagementContent.GET, 'engagement_id', String(engagementId));
+    if (!engagementId || isNaN(Number(engagementId))) {
+        return Promise.reject('Invalid Engagement Id ' + engagementId);
+    }
+    const response = await http.GetRequest<EngagementContent[]>(url); // Notice the change here
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply met-web/src/services/engagementContentService/index.ts]: Removed.

File: met-web/src/services/engagementSummaryService/index.ts (hunk 1 above)
Comment (VineetBala-AOT) [reply met-web/src/services/engagementSummaryService/index.ts]: Removed.

**Review by Baelx — APPROVED** [review]

**Pull request merged**